tasks/install-on-Darwin.yml

```diff
@@ -0,0 +1,29 @@
+---
+# https://buildkite.com/docs/agent/v3/osx
+# TODO:
+# * Conditionally upgrade when already present - `brew update && brew upgrade buildkite-agent`
+- homebrew_tap:
+    name: buildkite/buildkite
+    state: present
+
+- homebrew:
+    name: buildkite-agent
+    state: latest
+    update_homebrew: yes
+
+- name: Configure Buildkite
+  template:
+    src: "buildkite-agent.cfg.j2"
+    dest: "{{ buildkite_agent_conf_dir[ansible_os_family] }}/buildkite-agent.cfg"
+    mode: "0600"
+  notify:
+    - "Restart Buildkite Agent and Register for Login"
+
+- name: Set environment hook to load bash profile for test runs
+  template:
+    src: "buildkite-agent-hook-environment.sh"
+    dest: "{{ buildkite_agent_hooks_dir[ansible_os_family] }}/environment"
+    mode: "0600"
```

Comment:
What should owner and group be?

Reply:
whatever the installing user is, can be left implied

Reply:
That's true, but inconsistent with elsewhere in this role.

(I should say; I'm _not_ an authority of what lives in this repo, I don't have commit rights. My review here is trying to reduce the effort needed by the repo owner, but there's no style guide to refer to. So, apologies if subjective.)

Reply:
Second thought; if mode is 0600, for the agent user to read/execute, it will need to be either 0750 or owned and group of the agent and 0700.

It might contain secrets (that's a recommendation within the bk docs), so I would prefer 0700 and owned and grouped to agent user.

Reply:
@tommeier I just twigged to "whatever the installing user is, can be left implied" - I would deeply prefer that there be a user called `buildkite-agent` (that has no sudo etc privileges) that is the user the buildkite-agent process(es) are run by, compared to whichever user the playbook is executed by (since that user is quite likely to be quite privileged).

I think that can be achieved via [`user`](https://docs.ansible.com/ansible/2.7/modules/user_module.html) and then a combination of `become` and `become_user`.

If you'd prefer, I can do that - however, I'm unlikely to make it an optional thing since for my own agents, I'm very concerned about security (see buildkite/feedback#360). In that instance, I'd merge this PR as is, make a release (so you can pin, since presumably this is working for your own use-cases), and then at some point in future make the change and another (semver) release.

(cc @hectcastro)

Reply:
yup we need it to run native tests properly - so you're welcome to make this change - FYI this has been working perfectly over all our mac minis for the last month